rendermodules/rough_align/apply_rough_alignment_to_montages.py

```diff
@@ -0,0 +1,244 @@
+        # get lowres stack tile specs
+        lowres_ts = render.run(
+                            renderapi.tilespec.get_tile_specs_from_z,
+                            lowres_stack,
+                            newz)
+
+        # get input stack tile specs (for the first tile)
+        highres_ts = render.run(
+                            renderapi.tilespec.get_tile_specs_from_z,
+                            input_stack,
+                            z)[0]
+
+        # get the lowres stack rough alignment transformation
+        tforms = lowres_ts[0].tforms
+        d = tforms[-1].to_dict()
+        
+        dsList = d['dataString'].split()
+        v0 = float(dsList[0])*scale
+        v1 = float(dsList[1])*scale
+        v2 = float(dsList[2])*scale
+        v3 = float(dsList[3])*scale
+        v4 = float(dsList[4])
+        v5 = float(dsList[5])
+        d['dataString'] = "%f %f %f %f %s %s"%(v0, v1, v2, v3, v4, v5)
+        tforms[-1].from_dict(d)
+        stackbounds = render.run(
+                            renderapi.stack.get_bounds_from_z,
+                            input_stack,
+                            z)
+        prestackbounds = render.run(
+                            renderapi.stack.get_bounds_from_z,
+                            prealigned_stack,
+                            z)
+        
+        tx = 0
+        ty = 0
+        if input_stack == prealigned_stack:
+            tx =  -int(stackbounds['minX']) #- int(prestackbounds['minX'])
+            ty =  -int(stackbounds['minY']) #- int(prestackbounds['minY'])
+        else:
+            tx = int(stackbounds['minX']) - int(prestackbounds['minX'])
+            ty = int(stackbounds['minY']) - int(prestackbounds['minY'])
+        
+        tforms1 = highres_ts.tforms
+        d = tforms1[-1].to_dict()
+        
+        dsList = d['dataString'].split()
+        v0 = 1.0
+        v1 = 0.0
+        v2 = 0.0
+        v3 = 1.0
+        v4 = tx
+        v5 = ty
+        d['dataString'] = "%f %f %f %f %s %s"%(v0,v1,v2,v3,v4,v5)
+        tforms1[-1].from_dict(d)
+        
+        # delete the lens correction transform
+        if len(tforms1) > 1:
+            del tforms1[0]
+
+        ftform = tforms1 + tforms
+        
+        allts = []
+        highres_ts1 = render.run(
+                            renderapi.tilespec.get_tile_specs_from_z,
+                            input_stack,
+                            z)
+
+        for t in highres_ts1:
+            for f in ftform:
+                t.tforms.append(f)
+            if consolidateTransforms:
+                newt = consolidate_transforms(t.tforms)
+                t.tforms = newt
+            d1 = t.to_dict()
+            d1['z'] = newz
+            allts.append(d1)
+
+        tilespecfilename = os.path.join(output_dir,'tilespec_%04d.json'%newz)
+        fp = open(tilespecfilename,'w')
+        json.dump([ts for ts in allts], fp, indent=4)
+        fp.close()
+    except:
+        print("This z has not been aligned!")
```

Comment:
do we want this to proceed as if nothing happened if there is a z section that we ask to apply alignment to that has not been aligned?

Reply:
A potential case where you would want to simply pass is in case of a missing section... But if we are handling that in a different way then this will become an exception.

Reply:
the missing section case is a good use case to consider.. doesn't the module code detect though what z values are present at the start of the module?  Another suggestion would be to have the partial mapped function return True/False per section, and then have the output json report the z values of the sections that were successfully solved.  In that way the strategy could make a call as to whether it should proceed or not.

Reply:
Technically rough alignment shouldn't be affected if there is a missing section. Hence the reason why I left the exception to be a print statement... But returning true/false seems good, Although I feel that there might be more use cases like no point matches found, etc...

Reply:
agreed that bare try/except is scary.  Which exceptions are raised in this block -- RenderErrors? IOErrors?  It would be nice if we could differentiate between, say, a missing section and a failure to get stack bounds because either input_stack or prealigned_stack are in LOADING state.

Reply:
I guess its appropriate to do these checks before this function gets executed... There is definitely a possibility that the montage stack is in LOADING state in terms of how the workflow is designed....

Reply:
Shouldn't the only expected failure be in `renderapi.stack.get_tile_specs_from_z`? If so, could failures in those just return None so anything else weird will just fail?
Also, it looks like `highres_ts` is only being used as a proxy for `renderapi.transform.AffineModel` and fetching all tilespecs in z can be expensive with large collections.

Reply:
I am still concerned about this bare try/except beyond expected failures, what if somehow the render server goes down or fails to communicate due to no problem of this code at all.. or nfs goes down, or who knows what.  I don't think we want the workflow plowing forward, it will just make it harder to figure out what happened and to fix it up later.